### PR TITLE
Support C++14 syntax

### PR DESCRIPTION
**Pull request by dohzya**
This PR adds support for the [language-cpp14](https://atom.io/packages/language-cpp14) package

[Review on Reviewable](https://reviewable.io/reviews/atomlinter/linter-clang/109)

### CODE REVIEW

File: lib/main.js [hunk 1]
```diff
@@ -26,6 +26,10 @@ export default {
       type: "string",
       default: "-Wall -std=c++11"
     },
+    clangDefaultCpp14Flags: {
+      type: "string",
```
Comment (steelbrain) [inline lib/main.js]: how about
```js
clangDefaultCpp14Flags: {
  type: 'array',
  default: ['-Wall', '-std=c++14'],
  items: {
    type: 'string'
  }
}
```

File: lib/main.js [hunk 2]
```diff
@@ -87,6 +91,11 @@ export default {
           args.push("-xc++");
           args.push(...atom.config.get("linter-clang.clangDefaultCppFlags").split(/\s+/));
         }
+        if(activeEditor.getGrammar().name === "C++14") {
+          //const language = "c++";
+          args.push("-xc++");
+          args.push(...atom.config.get("linter-clang.clangDefaultCpp14Flags").split(/\s+/));
```
Comment (steelbrain) [inline lib/main.js]: You won't have to do this splitting and all if you make this an array, you can just concat then

Comment (dohzya) [inline lib/main.js]: I don't know if this method works with the settings page :-)

And even if it does, I think this change should be done in a dedicated PR ;-)

Comment (steelbrain) [inline lib/main.js]: It works in the settings page, trust me, I work with atom packages each day :)
You do the change I suggested above and then you'll get an array when you get this config.

File: lib/main.js (hunk 1 above)
Comment (k2b6s9j) [inline lib/main.js]: @steelbrain The reason it's a string that gets split by Space, it would look weird to users to have to separate parameters by comma.

Comment (steelbrain) [inline lib/main.js]: @k2b6s9j Oh no, I completely forgot about that.

File: lib/main.js (hunk 2 above)
Comment (k2b6s9j) [inline lib/main.js]: @dohzya Don't worry about it.

**Comment by k2b6s9j** [comment]
@dohzya The PR looks great. Small question though. Do know know if this will ever get merged with the C grammar?

**Comment by dohzya** [comment]
Honestly I have no idea, I found the C++14 package by chance but I don't know anything about it :-)

I hesitated merge `C++` and `C++14` conditions and replace the `-std=c++11` into `-std=c++14` in case of `C++14`, but this approach makes the behavior more obscure (the show default is not the real one), so I let this version with the 2 C++ versions separated.

Maybe a safer approach would be to accept `C++` and `C++14` (maybe `C++11`, `C++17`, just in case) but to do nothing with them? (let the `-std=c++11` by default)

**Comment by k2b6s9j** [comment]
In that case I'd feel better just adding an extra conditional to the code that loads C++ config than add a whole new config for a package with an unsure future.

**Comment by dohzya** [comment]
I put a new version which accept all grammars whose name starts by “C++”.

**Comment by k2b6s9j** [comment]
Okay. This should do the trick, this will be in the next patch or greater release.
